docs/src/examples.md

```diff
@@ -0,0 +1,609 @@
+```@contents
+Depth = 3
+```
+
+## API Reference
+```@docs
+CSV.read
+CSV.File
+CSV.Chunks
+CSV.Rows
+```
+
+## Utilities
+```@docs
+CSV.detect
+```
+
+## Examples
+
+### [Non-UTF-8 character encodings](@id stringencodings)
+
+```julia
+# assume I have csv text data encoded in ISO-8859-1 encoding
+# I load the StringEncodings package, which provides encoding conversion functionality
+using CSV, StringEncodings
+
+# I open my `iso8859_encoded_file.csv` with the `enc"ISO-8859-1"` encoding
+# and pass the opened IO object to `CSV.File`, which will read the entire
+# input into a temporary file, then parse the data from the temp file
+file = CSV.File(open("iso8859_encoded_file.csv", enc"ISO-8859-1"))
+
+# to instead have the encoding conversion happen in memory, pass
+# `buffer_in_memory=true`; this can be faster, but obviously results
+# in more memory being used rather than disk via a temp file
+file = CSV.File(open("iso8859_encoded_file.csv", enc"ISO-8859-1"); buffer_in_memory=true)
+```
+
+### [Gzipped input](@id gzipped_input)
+
+```julia
+# assume I have csv text data compressed via gzip
```

Comment:
I think it would be useful to tell the users:
* what compression formats would be automatically handled
* what extensions are automatically detected (unless it is solely contents based format detection - and if this is the case also comment on this)

Reply:
good call; added.